spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java

```diff
@@ -428,24 +428,22 @@ else if (!StringUtils.isEmpty(apps)) {
 	 * {@link org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader}.
 	 */
 	private void prefetchMetadata(List<AppRegistration> appRegistrations) {
-		forkJoinPool.execute(() -> {
-			appRegistrations.stream().filter(r -> r.getMetadataUri() != null).parallel().forEach(r -> {
-				logger.info("Eagerly fetching {}", r.getMetadataUri());
-				try {
-					this.appRegistryService.getAppMetadataResource(r);
-				}
-				catch (Exception e) {
-					logger.warn("Could not fetch " + r.getMetadataUri(), e);
-				}
-			});
-		});
+		forkJoinPool.execute(() -> appRegistrations.stream().filter(r -> r.getMetadataUri() != null).parallel().forEach(r -> {
+			logger.info("Eagerly fetching {}", r.getMetadataUri());
+			try {
+				this.appRegistryService.getAppMetadataResource(r);
+			}
+			catch (Exception e) {
+				logger.warn("Could not fetch {}", r.getMetadataUri(), e);
+			}
+		}));
 	}
 
 	private void validateApplicationName(String name) {
 
 		// Check for length of name to be less than 256 character.
 		if (name.length() > 255) {
-			throw new IllegalArgumentException("Length of application name must be less than 256 characters");
+			throw new InvalidApplicationNameException("Length of application name must be less than 256 characters");
```

Comment:
No test changes needed?

Reply:
Wanted to run the tests here as in local there was some issue, will resolve the tests.